exchanges/orderbook/orderbook_test.go

```diff
@@ -517,16 +568,136 @@ func TestProcessOrderbook(t *testing.T) {
 	wg.Wait()
 }
 
-func TestSetNewData(t *testing.T) {
-	err := service.SetNewData(nil, "")
+func deployUnorderedSlice() []Item {
+	var items []Item
+	rand.Seed(time.Now().UnixNano())
+	for i := 0; i < 100; i++ {
+		items = append(items, Item{Amount: 1, Price: rand.Float64(), ID: rand.Int63()}) // nolint:crypto/rand
+	}
+	return items
+}
+
+func TestSorting(t *testing.T) {
+	var b Base
+
+	b.Asks = deployUnorderedSlice()
+	err := b.Verify()
 	if err == nil {
-		t.Error("error cannot be nil")
+		t.Fatal("error cannot be nil")
+	}
+
+	SortAsks(nil)
+
+	SortAsks(b.Asks)
+	err = b.Verify()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	b.Bids = deployUnorderedSlice()
+	err = b.Verify()
+	if err == nil {
+		t.Fatal("error cannot be nil")
 	}
+
+	SortBids(nil)
+
+	SortBids(b.Bids)
+	err = b.Verify()
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+// 292588	      4785 ns/op	      32 B/op	       1 allocs/op
+func BenchmarkSortAsks(b *testing.B) {
+	s := deployUnorderedSlice()
+	for i := 0; i < b.N; i++ {
+		SortAsks(s)
+	}
+}
+
+// 272722	      4811 ns/op	      48 B/op	       2 allocs/op
+func BenchmarkSortBids(b *testing.B) {
+	s := deployUnorderedSlice()
+	for i := 0; i < b.N; i++ {
+		SortBids(s)
+	}
+}
+
+func deploySliceOrdered(size int) []Item {
+	rand.Seed(time.Now().UnixNano())
+	var items []Item
+	for i := 0; i < size; i++ {
+		items = append(items, Item{Amount: 1, Price: float64(i + 1), ID: rand.Int63()}) // nolint:crypto/rand
+	}
+	return items
 }
 
-func TestGetAssociations(t *testing.T) {
-	_, err := service.GetAssociations(nil, "")
+func TestReverse(t *testing.T) {
+	var b Base
+
+	length := 100
+	b.Bids = deploySliceOrdered(length)
+	if len(b.Bids) != length {
+		t.Fatal("incorrect length")
+	}
+
+	err := b.Verify()
 	if err == nil {
-		t.Error("error cannot be nil")
+		t.Fatal("error cannot be nil")
+	}
+
+	Reverse(nil)
+	Reverse(b.Bids)
+	err = b.Verify()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	b.Asks = append(b.Bids[:0:0], b.Bids...) // nolint:gocritic
+	err = b.Verify()
+	if err == nil {
+		t.Fatal("error cannot be nil")
+	}
+
+	Reverse(b.Asks)
+	err = b.Verify()
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+// 6217651	       187 ns/op	       0 B/op	       0 allocs/op
+func BenchmarkReverse(b *testing.B) {
+	length := 100
+	s := deploySliceOrdered(length)
+	if len(s) != length {
+		b.Fatal("incorrect length")
+	}
+
+	for i := 0; i < b.N; i++ {
+		Reverse(s)
```

Comment:
Please add another benchmark which includes verify. Since from looking at it, the inclusion of this makes verifying mandatory where a sorting a slice guarantees the results

Reply:
You need to verify both instances, especially duplications and zero values. All I was attempting to do is highlight the reduced overhead of using a reverse feature when a properly ascending or descending value is returned compared to sorting.